### PR TITLE
add intx

### PR DESCRIPTION
**Pull request by pwqbot**
Specify library name and version:  **intx/0.10.1**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/intx/all/test_package/test_package.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// intx: extended precision integer library.
+// Copyright 2019 Pawel Bylica.
+// Licensed under the Apache License, Version 2.0.
+
```
Comment (valgur) [inline recipes/intx/all/test_package/test_package.cpp:4]: Including a copyright notice on a single line of code seems a bit excessive to me.

**Review by valgur — APPROVED** [review]

File: recipes/intx/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    topics = ("evm", "biginteger", "arbitrary-precision", "header-only")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    def export_sources(self):
+        export_conandata_patches(self)
```
Comment (czoido) [inline recipes/intx/all/conanfile.py:26]: Hi @pwqbot, thanks for the contribution. This should not be needed if there are not patches.
```suggestion
```

File: recipes/intx/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        apply_conandata_patches(self)
+
```
Comment (czoido) [inline recipes/intx/all/conanfile.py:63]: Same as before, not needed if there are no patches.
```suggestion
```

File: recipes/intx/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            "msvc": "192",
+            "gcc": "12",
+            "clang": "15",
+            "apple-clang": "14.1",
+        }
+
+
+    def requirements(self):
+        pass
+
```
Comment (czoido) [inline recipes/intx/all/conanfile.py:44]: No need to declare this if there are no requirements.
```suggestion
```

**Comment by czoido** [comment]
> @czoido, this PR was merged by the bot despite an unexpected error causing the V2 pipeline to fail. Does this mean the recipe is ready for V2, or should I reopen this PR?
Hi @pwqbot, don't worry, no need to reopen, we can trigger the builds manually. Thanks for the contribution.

**Pull request merged**